core/trino-main/src/main/java/io/trino/metadata/TypeRegistrations.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.metadata;
+
+import io.trino.spi.block.BlockEncodingSerde;
+import io.trino.spi.type.TypeManager;
+import io.trino.type.JsonPath2016Type;
+import io.trino.type.TypeDeserializer;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+
+import static java.util.Objects.requireNonNull;
+
+public class TypeRegistrations
```

Comment:
would that make sense to move this to `json` subpackage. 
Also add a comment why we need to register `JsonPath2016Type` in `@PostConstruct` so it is obvious we are working around circular dependency of type<->typemanager here.

Reply:
I removed `TypeRegistrations`, and modelled it similarly to registering functions in `ServerMainModule`.